File: VaultNetAPI/VaultNet/Db/Change.cs
using ChargedChat.Db;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VaultNet.Db
{
    public class Change
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ChangeId { get; set; }

        [Required]
        public int ProductId {  get; set; }

        [ForeignKey(nameof(ProductId))]
        public Product Product { get; set; }

        public required string Changes {  get; set; }

        public int UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public  User User { get; set; }

    }
}

File: VaultNetAPI/src/main/java/com/ventthos/Vaultnet/domain/Change.java
package com.ventthos.Vaultnet.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "changes")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Change {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long changeId;

    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private String changes;
}
